presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -106,12 +128,32 @@ public ResourceGroupInfo getResourceGroupInfo(ResourceGroupId id)
         return groups.get(id).getPathToRoot();
     }
 
+    private boolean shouldQueueQueries()
+    {
+        return (getActiveWorkerCount() < initializationRequiredWorkers && shouldQueueQueriesWithInsufficientWorkers);
+    }
+
+    private int getActiveWorkerCount()
+    {
+        int activeWorkerCount = internalNodeManager.getNodes(ACTIVE).size();
+        if (!isIncludeCoordinator) {
+            activeWorkerCount--;
+        }
+        return activeWorkerCount;
+    }
     @Override
     public void submit(Statement statement, QueryExecution queryExecution, SelectionContext<C> selectionContext, Executor executor)
     {
         checkState(configurationManager.get() != null, "configurationManager not set");
+        int activeWorkerCount = getActiveWorkerCount();
+        if (!acceptQueries.get() && !shouldQueueQueriesWithInsufficientWorkers) {
```

Comment:
Same as above -- Pending all queries infinitely might be hiding another issue in the cluster